app/src/main/java/com/example/habithelper/adapters/HabitAdapter.java

```diff
@@ -0,0 +1,97 @@
+package com.example.habithelper.adapters;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.core.app.ActivityOptionsCompat;
+import androidx.core.util.Pair;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.habithelper.R;
+import com.example.habithelper.activities.HabitDetailsActivity;
+import com.example.habithelper.models.Habit;
+
+import java.sql.SQLOutput;
+import java.util.List;
+import org.parceler.Parcels;
+
+public class HabitAdapter extends RecyclerView.Adapter<HabitAdapter.ViewHolder> {
+
+    private Activity mActivity;
+    private Context context;
+    private List<Habit> habitList;
+
+    public HabitAdapter(Activity mActivity, Context context, List<Habit> habitList){
+        this.mActivity = mActivity;
+        this.context = context;
+        this.habitList = habitList;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.item_habit, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        Habit habit = habitList.get(position);
+        holder.bind(habit);
+    }
+
+    @Override
+    public int getItemCount() {
+        return habitList.size();
+    }
+
+    public void addAll(List<Habit> list){
+        habitList.addAll(list);
+        notifyDataSetChanged();
+    }
+
+    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private TextView tvHabit;
+        private ImageView ivHabit;
+
+        public ViewHolder(@NonNull View itemView) {
+            super(itemView);
+            ivHabit = itemView.findViewById(R.id.ivHabit);
+            tvHabit = itemView.findViewById(R.id.tvHabit);
+            itemView.setOnClickListener(this);
+        }
+
+        public void bind(Habit habit) {
+            tvHabit.setText(habit.getHabitName());
+            if(habit.get("habitImage") != null){
+                ivHabit.setBackground(AppCompatResources.getDrawable(context, (Integer) habit.getHabitImageKey()));
```

Comment:
this code looks duplicated from Habits Activity, any reason we have it in 2 places?

Reply:
This is in the adapter class that is displaying the recyclerView of habits, but the HabitDetailsActivity class is a detailed view of one habit. I have the habit being passed using Parcels, but so far, I haven't found a more elegant way of saving a drawable to Parse than saving the image key and obtaining and translating it locally. It's difficult because there is no drawable field on the Parse database. In my next commit, I have tried something new where I save the String name of the drawable which cuts down on some but not all of the repeated code!